xds/internal/balancer/edsbalancer/xds_client_wrapper.go

```diff
@@ -45,6 +47,43 @@ var (
 	bootstrapConfigNew = bootstrap.NewConfig
 )
 
+type loadStoreWrapper struct {
```

Comment:
Do we need this type?

If we cached the `PerClusterStore` in the `xdsClientWrapper` whenever the LRS stream is to be updated (either the client changed, or the LRS server name changed or the edsServiceName changed), then it will simplify things, wouldn't it?

Also, I feel that caching the `PerClusterStore` would reduce lock contention, and the `Store` can also be simplified by not having to require a `RWLock`.